### PR TITLE
Add more resource descriptions and convert resources to use properties

### PR DESCRIPTION
**Pull request by tas50**
More general resource cleanup. Add more resource descriptions and convert resources to use properties

### CODE REVIEW

**Comment by lamont-granquist** [comment]
integration tests on travis seem pissed, i don't understand what it has to do with this though

**Comment by lamont-granquist** [comment]
i can't replicate the failure on master on my desktop tho...

**Comment by tas50** [comment]
These failures are super odd

File: lib/chef/resource/template.rb [hunk 1]
```diff
@@ -60,29 +57,9 @@ def source(file = nil)
...
-      end
-
-      def local(args = nil)
-        set_or_return(
-          :local,
-          args,
-          :kind_of => [ TrueClass, FalseClass ]
-        )
-      end
+      property :variables, Hash, default: {}
```
Comment (lamont-granquist) [inline lib/chef/resource/template.rb]: this is a frozen hash default.

```ruby
template "whatever.erb" do
   variables[:aliases] ||= {}
end
```

That mutates the Hash which is the default value.

You need to get a new Hash object for every new template resource.

Frozen default values probably just caught a huge bug here.

Comment (lamont-granquist) [reply lib/chef/resource/template.rb]: `default: lazy { Hash.new }` maybe?

Comment (lamont-granquist) [reply lib/chef/resource/template.rb]: yeah validated that produces different objects.

i'd almost go so far as to say that `default: {}` is always a bug and should be replaced with this (and any default hash, even with default keys should be lazied this way).

**Pull request merged**